components/mediation-ui/org.wso2.carbon.message.processor.ui/src/main/java/org/wso2/carbon/message/processor/ui/MessageProcessorAdminServiceClient.java

```diff
@@ -469,15 +469,15 @@ public void popMessage(String processorName) throws Exception {
      *
      * @param processorName MessageProcessor name
      * @param storeName Name of store to redirect the message
+     * @return <code>true</code> if popAndRedirectMessage is successful, else <code>false</code>
      * @throws Exception
      */
-    public void popAndRedirectMessage(String processorName, String storeName) throws Exception {
+    public boolean popAndRedirectMessage(String processorName, String storeName) throws Exception {
         try {
-            if (processorName != null) {
-                stub.popAndRedirectMessage(processorName, storeName);
-            }
+            return stub.popAndRedirectMessage(processorName, storeName);
         } catch (Exception e) {
             handleException(e);
+            return false;
```

Comment:
We cha change here as well.

Reply:
added a new function createException() to fix the issue